ext/standard/crc32.c

```diff
@@ -35,6 +61,34 @@ PHP_NAMED_FUNCTION(php_if_crc32)
 
 	crc = crcinit^0xFFFFFFFF;
 
+#if defined(__aarch64__)
+	if (has_crc32_insn()) {
+		int16_t *p2;
+		int64_t *p8 = (int64_t *)p;
+
+		while(nr >= sizeof(int64_t)) {
+			crc = __crc32d(crc, *p8++);
+			nr -= sizeof(int64_t);
+		}
+		if (nr >= sizeof(int32_t)) {
+			int32_t *p4 = (int32_t *)p8;
+			crc = __crc32w(crc, *p4++);
+			nr -= sizeof(int32_t);
+			p2 = (int16_t *)p4;
+		} else {
+			p2 = (int16_t *)p8;
+		}
+		if (nr >= sizeof(int16_t)) {
+			crc = __crc32h(crc, *p2++);
+			nr -= sizeof(int16_t);
+		}
+		if (nr) {
+			p = (char *)p2;
+			crc = __crc32b(crc, *p);
+                        --nr;
```

Comment:
nit: Stray indentation.

Reply:
Thanks @nikic for the review.
I just pushed a corrected version with all three cases looking the same.